scipy/stats/tests/test_stats.py

```diff
@@ -7652,12 +7720,11 @@ def test_length0_2d_error(self):
             assert_equal(p, nans)
 
     def test_all_length_one(self):
-        msg = 'all input arrays have length 1.'
-        with pytest.warns(stats.DegenerateDataWarning, match=msg):
+        with pytest.warns(SmallSampleWarning):
             result = stats.f_oneway([10], [11], [12], [13])
             assert_equal(result, (np.nan, np.nan))
 
-    @pytest.mark.parametrize('args', [(), ([1, 2, 3],)])
+    @pytest.mark.parametrize('args', [([1, 2, 3],)])
```

Comment:
This is technically undesirable:
```python3
from scipy import stats
stats.f_oneway([])  # F_onewayResult(statistic=nan, pvalue=nan) and "too small" warning
stats.f_oneway([1, 2, 3])  # at least two inputs are required; got 1.
```
This probably affects all decorated functions that accepts `*args`. It happens because the decorator produces the warning and result before checking the number of arguments.

Reply:
I guess we leave this for a follow-up? If so, could you open an issue about it?

Reply:
Fixed.